cnf-tests/testsuites/e2esuite/ovs_qos/ovs_qos.go

```diff
@@ -0,0 +1,989 @@
+					qosNodeSelector, egressBitLimit, err = findQoSRateWithNodeSelector(isEgressLimiting, qosNodeSelector)
+					if err != nil {
+						eSkipReason = err.Error()
+						Skip(eSkipReason)
+					}
+				}
+
+				nodes, err := client.Client.Nodes().List(context.Background(), metav1.ListOptions{
+					LabelSelector: qosNodeSelector,
+				})
+
+				if err != nil {
+					eSkipReason = err.Error()
+					Fail(eSkipReason)
+				}
+
+				filtered, err := utilNodes.MatchingOptionalSelector(nodes.Items)
+				if err != nil {
+					eSkipReason = err.Error()
+					Fail(eSkipReason)
+				}
+
+				if len(filtered) <= 1 {
+					if discovery.Enabled() {
+						eSkipReason = "Did not find enough nodes with ovs_qos limitations"
+						Skip(eSkipReason)
+					} else {
+						eSkipReason = "Not enough nodes for ovs_qos tests"
+						Fail(eSkipReason)
+					}
+
+				}
+
+				receiverNode = filtered[0].ObjectMeta.Labels[utils.LabelHostname]
+				senderNode = filtered[1].ObjectMeta.Labels[utils.LabelHostname]
+
+				if !discovery.Enabled() {
+					mcpName, err = findMCPByMCRole(roleWorkerCNF)
+					if err != nil {
+						eSkipReason = err.Error()
+						Fail(err.Error())
+					}
+
+					_, egressBitLimit, err = findQoSRateWithNodeSelector(isEgressLimiting, qosNodeSelector)
+					if err != nil {
+						initialRate, err := findInitialRate(receiverNode, senderNode, true)
+						if err != nil {
+							namespaces.Clean(namespaces.OVSQOSTest, "testpod-", client.Client)
+							eSkipReason = err.Error()
+							Fail(eSkipReason)
+						}
+
+						egressBitLimit = initialRate * utils.LimitMultiplier
+						_, err = createEgressMC(int64(egressBitLimit))
+						if err != nil {
+							eSkipReason = err.Error()
+							Fail(eSkipReason)
+						}
+						mcpUpdated = true
+					}
+
+					if mcpUpdated {
+						err = waitForMCPStable(mcpName)
+						if err != nil {
+							eSkipReason = err.Error()
+							Fail(err.Error())
+						}
+					}
+				}
+
+			})
+
+			BeforeEach(func() {
+				if gSkipReason != "" {
+					Skip(gSkipReason)
+				}
+				if eSkipReason != "" {
+					Skip(eSkipReason)
+				}
+
+				namespaces.Clean(namespaces.OVSQOSTest, "testpod-", client.Client)
+			})
+
+			It("Validate MCO applied egress MachineConfig on the relevant nodes", func() {
+				err := validateEgressQosOnNodes(int64(egressBitLimit))
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			DescribeTable("Test limitations are correctly applied", func(connectivity string) {
+				By("Getting connectivity implied networks")
+				senderHostNetwork, receiverHostNetwork, err := networksByConnectivity(connectivity)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Creating receiver pod")
+				receiverPod := createReceiverPod(receiverNode, receiverHostNetwork)
+				receiverIP := receiverPod.Status.PodIP
+
+				By("Creating sender pod")
+				senderPod := createSenderPod(senderNode, senderHostNetwork, receiverIP)
+
+				By("Waiting for the sender to end")
+				time.Sleep(15 * time.Second)
```

Comment:
Can we add here Eventually loop which will wait until sender pod is in Completed state? In additional we has to remove "Sleep INF" command from sender pod

Reply:
done